src/core/qgsabstractgeopdfexporter.cpp

```diff
@@ -253,31 +256,78 @@ QString QgsAbstractGeoPdfExporter::createCompositionXml( const QList<ComponentLa
   // layertree
   QDomElement layerTree = doc.createElement( QStringLiteral( "LayerTree" ) );
   //layerTree.setAttribute( QStringLiteral("displayOnlyOnVisiblePages"), QStringLiteral("true"));
-  QSet< QString > createdLayerIds;
+  QMap< QString, QSet< QString > > createdLayerIds;
+  QMap< QString, QDomElement > groupLayerMap;
   if ( details.includeFeatures )
   {
     for ( const VectorComponentDetail &component : qgis::as_const( mVectorComponents ) )
     {
       QDomElement layer = doc.createElement( QStringLiteral( "Layer" ) );
-      layer.setAttribute( QStringLiteral( "id" ), component.mapLayerId );
+      layer.setAttribute( QStringLiteral( "id" ), component.group.isEmpty() ? component.mapLayerId : QStringLiteral( "%1_%2" ).arg( component.group, component.mapLayerId ) );
       layer.setAttribute( QStringLiteral( "name" ), component.name );
       layer.setAttribute( QStringLiteral( "initiallyVisible" ), QStringLiteral( "true" ) );
-      layerTree.appendChild( layer );
-      createdLayerIds.insert( component.mapLayerId );
+
+      if ( !component.group.isEmpty() )
+      {
+        if ( groupLayerMap.contains( component.group ) )
+        {
+          groupLayerMap[ component.group ].appendChild( layer );
+        }
+        else
+        {
+          QDomElement group = doc.createElement( QStringLiteral( "Layer" ) );
+          group.setAttribute( QStringLiteral( "id" ), QStringLiteral( "group_%1" ).arg( component.group ) );
+          group.setAttribute( QStringLiteral( "name" ), component.group );
+          group.setAttribute( QStringLiteral( "initiallyVisible" ), groupLayerMap.empty() ? QStringLiteral( "true" ) : QStringLiteral( "false" ) );
+          group.setAttribute( QStringLiteral( "mutuallyExclusiveGroupId" ), QStringLiteral( "__mutually_exclusive_groups__" ) );
```

Comment:
It's nice PDF spec allows for mutually exclusive groups.

Reply:
Yep :) There's actually a lot of other nice stuff possible here (but which can wait for future releases)